### DISCUSSION TITLE
Role's permissions reset on redeployment

### DISCUSSION THREAD

**Discussion by Robindeep-Kaur** [apache/airflow#36965] [Q&A] [answered]
I've deployed Airflow version 2.7.0 on Kubernetes k8s cluster. I'm using LDAP configuration to map roles and separately hitting roles API to create and update roles.
The role creation with initial permission is working fine. But when I try to update the role (UI as well as API call), it updates the role for a time being and as soon as I redeploy the Airflow instance on k8s, the role resets to the initial state/permissions.
Is there any sync parameter I can set/unset to resolve the above problem?

**Comment by potiuk** [comment, marked as answer]
This is more of a question to Flask App Builder that we use for authentication - so you can ask the question there, but I belive that syncing is happening by default and you are either supposed to use LDAP (and use it as the source of truth) or you should use API/UI. If you use both, then obviously you will get into trouble because you have no idea what is the "source of truth". As far as I know the LDAP configuration synchronizes both - users and roles. Maybe you can disable it - but ask in FAB
